exchanges/okx/okx_test.go

```diff
@@ -0,0 +1,3129 @@
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.GetMaximumWithdrawals(context.Background(), "BTC"); err != nil {
+		t.Error("Okx GetMaximumWithdrawals() error", err)
+	}
+}
+
+func TestGetAccountRiskState(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	// Operation is not supported under the current account mode
+	if _, err := ok.GetAccountRiskState(context.Background()); err != nil && !strings.Contains(err.Error(), "51010") {
+		t.Error("Okx GetAccountRiskState() error", err)
+	}
+}
+
+func TestVIPLoansBorrowAndRepay(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.VIPLoansBorrowAndRepay(context.Background(), LoanBorrowAndReplayInput{Currency: "BTC", Side: "borrow", Amount: 12}); err != nil &&
+		!strings.Contains(err.Error(), "Your account does not support VIP loan") {
+		t.Error("Okx VIPLoansBorrowAndRepay() error", err)
+	}
+}
+
+func TestGetBorrowAndRepayHistoryForVIPLoans(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.GetBorrowAndRepayHistoryForVIPLoans(context.Background(), "", time.Time{}, time.Time{}, 3); err != nil {
+		t.Error("Okx GetBorrowAndRepayHistoryForVIPLoans() error", err)
+	}
+}
+
+func TestGetBorrowInterestAndLimit(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.GetBorrowInterestAndLimit(context.Background(), 1, "BTC"); err != nil && !strings.Contains(err.Error(), "59307") { // You are not eligible for VIP loans
+		t.Error("Okx GetBorrowInterestAndLimit() error", err)
+	}
+}
+
+func TestPositionBuilder(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.PositionBuilder(context.Background(), PositionBuilderInput{
+		ImportExistingPosition: true,
+	}); err != nil {
+		t.Error("Okx PositionBuilder() error", err)
+	}
+}
+
+func TestGetGreeks(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.GetGreeks(context.Background(), ""); err != nil && !strings.Contains(err.Error(), "Unsupported operation") {
+		t.Error("Okx GetGreeks() error", err)
+	}
+}
+
+func TestGetPMLimitation(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.GetPMLimitation(context.Background(), "SWAP", "BTC-USDT"); err != nil {
+		t.Errorf("%s GetPMLimitation() error %v", ok.Name, err)
+	}
+}
+
+func TestViewSubaccountList(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.ViewSubAccountList(context.Background(), true, "", time.Time{}, time.Time{}, 1); err != nil {
+		t.Error("Okx ViewSubaccountList() error", err)
+	}
+}
+
+func TestResetSubAccountAPIKey(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := ok.ResetSubAccountAPIKey(context.Background(), &SubAccountAPIKeyParam{
```

Comment:
Please add a valid test case, even if it errors for you. I want to see if make a request to OKx